src/Sentry/Laravel/Tracing/Integrations/LighthouseIntegration.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Integrations;
+
+use GraphQL\Language\AST\OperationDefinitionNode;
+use Illuminate\Contracts\Events\Dispatcher;
+use Nuwave\Lighthouse\Events\EndExecution;
+use Nuwave\Lighthouse\Events\EndRequest;
+use Nuwave\Lighthouse\Events\StartExecution;
+use Nuwave\Lighthouse\Events\StartRequest;
+use Sentry\Laravel\Integration;
+use Sentry\SentrySdk;
+use Sentry\Tracing\SpanContext;
+
+class LighthouseIntegration
+{
+    /** @var array<int, <string|null, \GraphQL\Language\AST\OperationDefinitionNode>> $operations */
+    private $operations;
+
+    /** @var \Sentry\Tracing\Span|null $previousSpan */
+    private $previousSpan;
+
+    /** @var \Sentry\Tracing\Span|null $requestSpan */
+    private $requestSpan;
+
+    /** @var \Sentry\Tracing\Span|null $operationSpan */
+    private $operationSpan;
+
+    public function __construct(Dispatcher $evenDispatcher)
+    {
+        if (!class_exists(StartRequest::class)) {
```

Comment:
Would it make sense to do this detection before even instantiating the Integration? For coherence, the main service provider could call a static method within this class.

Reply:
I actually had that, and that might still be the better option. I'm not a 100% sure, this class exists feels a little too brittle too for my liking... some version check would be a lot better IMO but I'm not sure we can do that without adding dependencies or parsing the composer.json... still thinking about this 🧐

Reply:
I was thinking if we should add a canonical way of detecting if Lighthouse is installed to Lighthouse itself. On the other hand, this duck typing approach works nicely as it tests for the exact feature that is required.

Reply:
Also true, although it won't account for if in a major version of Ligthouse the even gains or loses properties we depend on, so we either need to program very defensively for that (this integration should _never_ crash) or we need to make sure we are on the correct version.

This is of course not limited to Ligthouse but to any integration we might add in the future.